Example/GiniVision/NoResultViewController.swift

```diff
@@ -19,6 +19,10 @@ class NoResultViewController: UIViewController {
     }
     
     @IBAction func retry(_ sender: AnyObject) {
-        _ = self.navigationController?.popToRootViewController(animated: true)
+        if let navVC = navigationController, !isFirstViewController(inNavController: navVC) {
```

Comment:
Another example of a view controller that knows too much about how it is presented. Although, here I might let it slide because at least it will work in most of the types of presentation.